src/arrayMethodJoin.js

```diff
@@ -5,8 +5,45 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```

Comment:
```suggestion
  [].__proto__.join2 = function(separator) {
```
let's add a default value to the `separator` argument (then there will be no need in your `checkSeparator` function